tiledb/sm/c_api/tiledb.cc

```diff
@@ -2364,12 +2369,64 @@ int32_t tiledb_array_delete_fragments(
   return TILEDB_OK;
 }
 
+capi_return_t tiledb_array_delete_fragments_v2(
+    tiledb_ctx_t* ctx,
+    const char* uri_str,
+    uint64_t timestamp_start,
+    uint64_t timestamp_end) {
+  auto uri = tiledb::sm::URI(uri_str);
+  if (uri.is_invalid()) {
+    throw api::CAPIStatusException(
+        "Failed to delete fragments; Invalid input uri");
+  }
+
+  // Allocate an array object
+  tiledb_array_t* array = new (std::nothrow) tiledb_array_t;
+  try {
+    array->array_ =
+        make_shared<tiledb::sm::Array>(HERE(), uri, ctx->storage_manager());
+  } catch (...) {
+    delete array;
+    array = nullptr;
+    throw api::CAPIStatusException("Failed to create array");
+  }
+
+  // Set array open timestamps
+  array->array_->set_timestamp_start(timestamp_start);
+  array->array_->set_timestamp_end(timestamp_end);
+
+  // Open the array for exclusive modification
+  throw_if_not_ok(array->array_->open(
+      static_cast<tiledb::sm::QueryType>(TILEDB_MODIFY_EXCLUSIVE),
+      static_cast<tiledb::sm::EncryptionType>(TILEDB_NO_ENCRYPTION),
+      nullptr,
+      0));
+
+  // Delete fragments
+  try {
+    array->array_->delete_fragments(uri, timestamp_start, timestamp_end);
+  } catch (...) {
+    throw_if_not_ok(array->array_->close());
+    delete array;
+    array = nullptr;
```

Comment:
Should we try to do nested exception here? I think the user might loose context of the exception.